api/dbadapters/bigquery.ts

```diff
@@ -96,22 +98,57 @@ export class BigQueryDbAdapter implements IDbAdapter {
       .promise();
   }
 
-  public async evaluate(statement: string) {
-    try {
-      await this.getClient().query({
-        useLegacySql: false,
-        query: statement,
-        dryRun: true
-      });
-      return dataform.QueryEvaluationResponse.create({
-        status: dataform.QueryEvaluationResponse.QueryEvaluationStatus.SUCCESS
-      });
-    } catch (e) {
-      return dataform.QueryEvaluationResponse.create({
-        status: dataform.QueryEvaluationResponse.QueryEvaluationStatus.FAILURE,
-        error: parseBigqueryEvalError(e)
-      });
+  public async evaluate(
+    queryOrAction: string | dataform.Table | dataform.Operation | dataform.Assertion,
+    projectConfig: dataform.IProjectConfig
+  ) {
+    let executionTasks: dataform.ExecutionTask[];
+    if (typeof queryOrAction === "string") {
+      executionTasks = [dataform.ExecutionTask.create({ statement: queryOrAction })];
+    } else {
+      try {
+        const coreAdapter = new BigQueryAdapter(projectConfig, version);
+        if (queryOrAction instanceof dataform.Table) {
+          executionTasks = coreAdapter
+            .publishTasks(
```

Comment:
just pinged you on chat, but this changes the semantics of evaluation a bit (sorry for just noticing this part).

i don't think we want to evaluate what is effectively an `ExecutionAction`, at least not in that part of the UI. I think a solution to that should be separate